sql-statements: readability of commit/rollback (#3664)
pingcap/docs: sql-statements/sql-statement-start-transaction.md

## MySQL compatibility

> * `START TRANSACTION` immediately starts a transaction inside TiDB. This differs from MySQL, where `START TRANSACTION` lazily creates a transaction unless the varation of `START TRANSACTION WITH CONSISTENT SNAPSHOT` is used.

Reviewer: To make it easier to understand, if my suggestion is what you mean.
```suggestion
* `START TRANSACTION` immediately starts a transaction inside TiDB. This differs from MySQL, where `START TRANSACTION` lazily creates a transaction. But `START TRANSACTION` in TiDB is equivalent to MySQL's `START TRANSACTION WITH CONSISTENT SNAPSHOT`.
```

> * The statement `START TRANSACTION READ ONLY` is parsed for compatibility with MySQL, but will still permit write operations.

Reviewer: ```suggestion
* The statement `START TRANSACTION READ ONLY` is parsed for compatibility with MySQL, but still allows write operations.
```
